src/NosCore.Controllers/GroupPacketController.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using NosCore.Configuration;
+using NosCore.GameObject;
+using NosCore.GameObject.Networking;
+using NosCore.Packets.ServerPackets;
+using NosCore.Shared.Enumerations.Character;
+using NosCore.Shared.Enumerations.Group;
+using NosCore.Shared.I18N;
+
+namespace NosCore.Controllers
+{
+    public class GroupPacketController : PacketController
+    {
+        private readonly WorldConfiguration _worldConfiguration;
+
+        [UsedImplicitly]
+        public GroupPacketController()
+        {
+            
+        }
+
+        public GroupPacketController(WorldConfiguration worldConfiguration)
+        {
+            _worldConfiguration = worldConfiguration;
+        }
+
+        /// <summary>
+        ///     pjoin packet
+        /// </summary>
+        /// <param name="pjoinPacket"></param>
+        public void GroupJoin(PjoinPacket pjoinPacket)
+        {
+            var targetSession = ServerManager.Instance.Sessions.Values.FirstOrDefault(s => s.Character.CharacterId == pjoinPacket.CharacterId);
+
+            if (targetSession == null && pjoinPacket.RequestType != GroupRequestType.Sharing)
+            {
+                return;
+            }
+
+            switch (pjoinPacket.RequestType)
+            {
+
+                case GroupRequestType.Requested:
+                case GroupRequestType.Invited:
+                    if (pjoinPacket.CharacterId <= 0 || targetSession == null || pjoinPacket.CharacterId == Session.Character.CharacterId)
```

Comment:
put a unsigned in the packet. validation should come from packets attributes when possible.